core/services/gateway/handlers/functions/orm.go

```diff
@@ -16,24 +17,28 @@ import (
 )
 
 //go:generate mockery --quiet --name ORM --output ./mocks/ --case=underscore
-
 type ORM interface {
 	GetSubscriptions(offset, limit uint, qopts ...pg.QOpt) ([]CachedSubscription, error)
 	UpsertSubscription(subscription CachedSubscription, qopts ...pg.QOpt) error
+
+	GetAllowedSenders(offset, limit uint, qopts ...pg.QOpt) ([]common.Address, error)
```

Comment:
It probably doesn't make sense to keep all the methods together in one ORM object. They are completely separate - they use different DB tables, APIs are not related to each other and they are created independently for different purposes. @KuphJr already pointed out that we are even creating two instances of the same object. Why not have two separate ORM interfaces and implementations?

Reply:
I've grouped them in order not to duplicate the type, New method and dependencies while still keep consistency with other implementations of orms in which all of them are returning interfaces.

I do agree it would be better to separate the interfaces given they are logically different and non-dependent. what do you think about "accept interfaces, return structs" here? If we return a `FunctionsORM` struct that implements both interfaces:

```
type SubscriptionsORM interface {
	GetSubscriptions(offset, limit uint, qopts ...pg.QOpt) ([]CachedSubscription, error)
	UpsertSubscription(subscription CachedSubscription, qopts ...pg.QOpt) error
}

type AllowlistORM interface {
	GetAllowedSenders(offset, limit uint, qopts ...pg.QOpt) ([]common.Address, error)
	CreateAllowedSenders(allowedSenders []common.Address, qopts ...pg.QOpt) error
}
```
we would only need a type and constructor, because both have the same dependencies (*sqlx.DB, logger and routerContractAddress), but we would have individual interfaces.

Reply:
as discussed I've split the interface in two separate ORM interfaces each belonging to the corresponding pkg (allowlist, subscriptions)